camel-core/src/main/java/org/apache/camel/model/RouteDefinition.java

```diff
@@ -1132,6 +1169,17 @@ protected RouteContext addRoutes(CamelContext camelContext, Collection<Route> ro
             }
         }
 
+        // configure Log EIP mask
+        if (logMask != null) {
+            Boolean isLogMask = CamelContextHelper.parseBoolean(camelContext, getLogMask());
+            if (isLogMask != null) {
+                routeContext.setLogMask(isLogMask);
+                if (isLogMask) {
+                    log.debug("Security mask for Logging is enabled on route: {}", getId());
```

Comment:
Ah sorry my bad about the RuntimeConfiguration being two levels. So lets keep this as-is. I will get this merged now. Then you can work on the spring boot support next.